scripts/performance/measure-performance.js

```diff
@@ -0,0 +1,195 @@
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+// @ts-check
+const puppeteer = require('puppeteer');
+const fs = require('fs');
+const resolve = require('path').resolve;
+const path = resolve('./workspace');
+const profilesPath = './profiles/';
+
+const lcp = 'Largest Contentful Paint (LCP)';
+const performanceTag = braceText('Performance');
+
+let name = 'Measurement';
+let url = 'http://localhost:3000/#' + path;
+let folder = 'profile';
+let headless = true;
+let runs = 10;
+
+(async () => {
+    let defaultUrl = true;
+    const args = process.argv.slice(2);
+
+    for (let index = 0; index < args.length; index++) {
+        if (args[index].startsWith('--')) {
+            if (args[index] === '--headful') {
+                headless = false;
+                continue;
+            }
+            let next = args[index + 1];
+            if (next !== undefined && !next.startsWith('--')) {
+                if (args[index] === '--name') {
+                    name = next;
+                    index++;
+                    continue;
+                }
+                if (args[index] === '--url') {
+                    url = next;
+                    defaultUrl = false;
+                    index++;
+                    continue;
+                }
+                if (args[index] === '--folder') {
+                    folder = next;
+                    index++;
+                    continue;
+                }
+                if (args[index] === '--runs') {
+                    runs = parseInt(next);
+                    index++;
+                    continue;
+                }
+            }
+        }
+    }
+
+    if (defaultUrl) getOrCreateFolder(path);
+    getOrCreateFolder(profilesPath);
+    const folderPath = profilesPath + folder;
+    getOrCreateFolder(folderPath);
+    // Wait until the browser page is available
+    const browser = await puppeteer.launch({ headless: true });
+    const page = await browser.newPage();
+    await retryConnection(() => page.goto(url), 1000);
+    browser.close();
+
+    await measurePerformance(name, url, folderPath, headless, runs);
+})();
+
+async function measurePerformance(name, url, folder, headless, runs) {
+    const multipleRuns = runs > 1 ? true : false;
+    let runNr = 1;
+    let durations = [];
+    while (runs > 0) {
+        const browser = await puppeteer.launch({ headless: headless });
+        const page = await browser.newPage();
+
+        const file = folder + '/' + runNr + '.json';
+        await page.tracing.start({ path: file, screenshots: true });
```

Comment:
Do we need to write the file out or can we just use what we get back from page.tracing.stop()?

Reply:
The advantage of writing the file out is, that you can then also use that file to have a detailed look of the tracing inside of the Google DevTools using this file. This can give additional information as well as a good overview over the applications startup.